### PR TITLE
feat(fx-2512): exposes artists connection; supports letter argument

### PR DESCRIPTION
**Pull request by dzucconi**
Closes: [FX-2512](https://artsyproduct.atlassian.net/browse/FX-2512)

### CODE REVIEW

File: src/schema/v2/schema.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ const rootFields = {
   artworks: Artworks,
   artist: Artist,
   artists: Artists,
+  artistsByLetterConnection,
```
Comment (mzikherman) [inline src/schema/v2/schema.ts:118]: what do you think if this was just `artistsConnection`? Encoding the `...ByLetter` part feels like it's specific to an argument, whereas a more general purpose `artistsConnection` (which can expand in the future if needed for other args) feels maybe better.

**Review by mzikherman — APPROVED** [review]
Nice! Could use a little specc-y. Feel free to merge when it's safe to do so (the staging k8s migration is in progress atm).

File: src/schema/v2/schema.ts (hunk 1 above)
Comment (dzucconi) [reply src/schema/v2/schema.ts:118]: 👍

**Comment by dzucconi** [comment]
Lemme double check the artistsLoader in v1. (I think disabling the project wide search in that piece of the codebase is very troubling when trying to refactor)

File: src/schema/v2/__tests__/artists.test.ts [hunk 2]
```diff
@@ -74,6 +96,189 @@ describe("Artists", () => {
     `
 
     const { artists } = await runQuery(query, { artistLoader })
+
     expect(artists[0].slug).toEqual("andy-warhol")
+    expect(artists[1].slug).toEqual("pablo-picasso")
+  })
+})
+
+fdescribe("artistsConnection", () => {
```
Comment (damassi) [inline src/schema/v2/__tests__/artists.test.ts:105]: (fdescribe)

Comment (dzucconi) [reply src/schema/v2/__tests__/artists.test.ts:105]: I do this all the time. Going to enable https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-focused-tests.md 😅

**Comment by dzucconi** [comment]
@mzikherman can you give this another once over?

File: src/schema/v1/artists.ts [hunk 3]
```diff
@@ -51,7 +51,7 @@ const Artists: GraphQLFieldConfig<void, ResolverContext> = {
       )
     }
 
-    return artistsLoader(options)
+    return artistsLoader(options).then(({ body }) => body)
```
Comment (mzikherman) [inline src/schema/v1/artists.ts:54]: nice, this is a gotcha that always pops up!

File: src/schema/v2/artists.ts [hunk 4]
```diff
@@ -51,7 +59,99 @@ const Artists: GraphQLFieldConfig<void, ResolverContext> = {
...
+    },
+    letter: { type: GraphQLString },
+    page: { type: GraphQLInt },
+    size: { type: GraphQLInt },
+    sort: ArtistSorts,
+  }),
+  description: "A list of artists",
+  resolve: async (
+    _root,
+    args: {
```
Comment (mzikherman) [inline src/schema/v2/artists.ts:91]: wow typing `args` 😻

**Review by mzikherman — APPROVED** [review]
Nice schema addition!

**Pull request merged**